### PR TITLE
Some Refactoring

### PR DESCRIPTION
**Pull request by marload**
We performed some code refactoring. This refactoring can result in more readable and non-duplicate code.

Thanks!

### CODE REVIEW

File: packages/vaex-ml/vaex/ml/linear_model.py [hunk 1]
```diff
@@ -66,8 +65,7 @@ def fit(self, dataset, y_expression, progress=False):
             def coordinates(expression, limits, shape):
                 if dataset.is_category(expression):
                     return np.arange(dataset.category_count(expression))
-                else:
-                    return dataset.bin_centers(expression, limits, shape)
+                return dataset.bin_centers(expression, limits, shape)
```
Comment (maartenbreddels) [inline packages/vaex-ml/vaex/ml/linear_model.py:68]: matter of taste, some would convert this back :)

File: packages/vaex-ml/vaex/ml/transformations.py [hunk 2]
```diff
@@ -432,11 +432,11 @@ def transform(self, df):
 
         copy = df.copy()
 
-        for i in range(len(self.features)):
-            name = self.prefix + self.features[i]
+        for i, feature in self.features:
```
Comment (maartenbreddels) [inline packages/vaex-ml/vaex/ml/transformations.py:435]:
```suggestion
        for i, feature in enumerate(self.features):
```

File: packages/vaex-ml/vaex/ml/transformations.py [hunk 3]
```diff
@@ -492,9 +492,9 @@ def transform(self, df):
         '''
 
         copy = df.copy()
-        for i in range(len(self.features)):
-            name = self.prefix + self.features[i]
-            expr = copy[self.features[i]]
+        for i, feature in self.features:
```
Comment (maartenbreddels) [inline packages/vaex-ml/vaex/ml/transformations.py:495]:
```suggestion
        for i, feature in enumerate(self.features):
```

File: packages/vaex-ml/vaex/ml/transformations.py [hunk 4]
```diff
@@ -564,13 +564,13 @@ def transform(self, df):
         '''
 
         copy = df.copy()
-        for i in range(len(self.features)):
-            name = self.prefix+self.features[i]
-            expr = copy[self.features[i]]
+        for i, feature in self.features:
```
Comment (maartenbreddels) [inline packages/vaex-ml/vaex/ml/transformations.py:567]:
```suggestion
        for i, feature in enumerate(self.features):
```

File: packages/vaex-ml/vaex/ml/linear_model.py (hunk 1 above)
Comment (marload) [reply packages/vaex-ml/vaex/ml/linear_model.py:68]: Personally, I think that if syntax should be simplified in Early Return Pattern. Related comments are in this link.

https://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html

File: packages/vaex-ml/vaex/ml/transformations.py [hunk 5]
```diff
@@ -564,13 +564,13 @@ def transform(self, df):
...
-            expr = copy[self.features[i]]
+        for i, feature in enumerate(self.features):
+            name = self.prefix+feature
+            expr = copy[feature]
             if self.with_centering:
-                expr = expr - self.center_[i]
+                expr -= self.center_[i]
             if self.with_scaling:
-                expr = expr / self.scale_[i]
+                expr /= self.scale_[i]
```
Comment (JovanVeljanoski) [inline packages/vaex-ml/vaex/ml/transformations.py:573]: This is also a matter of taste, and what you are used to writing or looking at.

**Comment by maartenbreddels** [comment]
Thanks!

**Pull request merged**